src/test/java/com/onlinebookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,224 @@
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.onlinebookstore.dto.cartitem.CartItemRequestDto;
+import com.onlinebookstore.dto.cartitem.CartItemResponseDto;
+import com.onlinebookstore.dto.cartitem.CartItemUpdateDto;
+import com.onlinebookstore.dto.shoppingcart.ShoppingCartResponseDto;
+import com.onlinebookstore.model.Role;
+import com.onlinebookstore.model.User;
+import jakarta.annotation.Resource;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.sql.DataSource;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+class ShoppingCartControllerTest {
+    protected static MockMvc mockMvc;
+
+    @Resource
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired DataSource dataSource,
+            @Autowired WebApplicationContext applicationContext
+    ) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource(
+                            "database/add-users-carts-items-before-testing.sql")
+            );
+        }
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(
+                    connection,
+                    new ClassPathResource("database/remove-users-carts-items-after-testing.sql")
+            );
+        }
+    }
+
+    @Order(1)
+    @Test
+    @DisplayName("Get shopping cart")
+    void getShoppingCart_ValidRequest_Success() throws Exception {
+        // Given
+        User mockUser = getMockUser();
```

Comment:
you can use annotation
![image](https://github.com/andrey-kiyas/OnlineBookStore/assets/20326858/c552d908-0515-4d34-bb09-67678eb7cfd5)

Reply:
Yes, this is also possible, but I thought my version was more readable and simpler.